lib/kb_SetUtilities/kb_SetUtilitiesImpl.py

```diff
@@ -589,21 +596,10 @@ def KButil_Merge_FeatureSet_Collection(self, ctx, params):
                 'text_message': report
             }
 
-        reportName = 'kb_SetUtilities_merge_featureset_report_' + str(uuid.uuid4())
-        ws = workspaceService(self.workspaceURL, token=ctx['token'])
-        report_obj_info = ws.save_objects({'workspace': params['workspace_name'],
-                                           'objects': [{'type': 'KBaseReport.Report',
-                                                        'data': reportObj,
-                                                        'name': reportName,
-                                                        'meta': {},
-                                                        'hidden': 1,
-                                                        'provenance': provenance}]})[0]
-
-        # Build report and return
-        self.log(console, "BUILDING RETURN OBJECT")
-        report_ref = "{}/{}/{}".format(report_obj_info[6], report_obj_info[0], report_obj_info[4])
-        returnVal = {'report_name': reportName,
-                     'report_ref': report_ref}
+        # Save report
+        report_info = self.reportClient.create({'report':reportObj, 'workspace_name':params['workspace_name']})
+
+        returnVal = { 'report_name': report_info['name'], 'report_ref': report_info['ref'] }
```

Comment:
this is repeated several times -- can you turn this into a method?

Reply:
I want to leave this because it makes clear without having to dig into a sub method what the return fields are.  I left it this way intentionally.

Reply:
it's identical each time, though -- I'd just do something like

```py

def createReportAndReturn(self, params, reportObj):
    report_info = self.reportClient.create({'report':reportObj, 'workspace_name':params['workspace_name']})
    return { 'report_name': report_info['name'], 'report_ref': report_info['ref'] }
```

```py
# save report
returnVal = self.createReportAndReturn(params, reportObj)
```

Reply:
no

Reply:
more repeated code => more cruft => more difficult to update, more difficult for other people to navigate, edit, and review

Reply:
do you have any idea how much bad code we have?  I have so much to get done.  each time I commit I have to wait for the tests to get run.  this is not supposed to be a code cleanup exercise.  I'm adding AMA functionality.  This is not the purpose of these reviews.  only showstoppers.  enough!

Reply:
I appreciate where you're coming from, but we have to compromise.

Reply:
it needs to go out